src/argilla_llama_index/llama_index_handler.py

```diff
@@ -574,42 +632,77 @@ def on_event_start(
         event_type: CBEventType,
         payload: Optional[Dict[str, Any]] = None,
         event_id: Optional[str] = None,
-        parent_id: Optional[str] = None,
-        **kwargs: Any,
-    ) -> None:
-        """Run handlers when an event starts."""
+        parent_id: str = None,
+    ) -> str:
+        """
+        Store event start data by event type.
+
+        Args:
+            event_type (CBEventType): The event type to store.
+            payload (Dict[str, Any], optional): The payload to store.
+            event_id (str, optional): The event id to store.
+            parent_id (str, optional): The parent id to store.
+
+        Returns:
+            str: The event id.
+        """
+
         event = CBEvent(event_type, payload=payload, id_=event_id)
         self.events_data[event_id].append(event)
 
+        return event.id_
+
     def on_event_end(
         self,
         event_type: CBEventType,
         payload: Optional[Dict[str, Any]] = None,
-        event_id: Optional[str] = None,
-        **kwargs: Any,
+        event_id: str = None,
     ) -> None:
-        """Run handlers when an event ends."""
+        """
+        Store event end data by event type.
+
+        Args:
+            event_type (CBEventType): The event type to store.
+            payload (Dict[str, Any], optional): The payload to store.
+            event_id (str, optional): The event id to store.
+        """
+
         event = CBEvent(event_type, payload=payload, id_=event_id)
         self.events_data[event_id].append(event)
 
-def _get_time_diff(
-    event_1_time_str: str, 
-    event_2_time_str: str
-) -> float:
-    """Get the time difference between two events."""
+
+# Auxiliary methods
+def _get_time_diff(event_1_time_str: str, event_2_time_str: str) -> float:
```

Comment:
Maybe these can be moved to a helpers.py to avoid duplicating the code, also in the tests?